tests/api.c

```diff
@@ -13768,6 +13821,742 @@ static int test_wc_ecc_is_valid_idx (void)
+    byte        data[] = "Test data to encode.";
+
+    #if defined(USE_CERT_BUFFERS_2048)
+        byte        key[sizeof_client_key_der_2048];
+        byte        cert[sizeof_client_cert_der_2048];
+        word32      keySz = (word32)sizeof(key);
+        word32      certSz = (word32)sizeof(cert);
+        XMEMSET(key, 0, keySz);
+        XMEMSET(cert, 0, certSz);
+        XMEMCPY(key, client_key_der_2048, keySz);
+        XMEMCPY(cert, client_cert_der_2048, certSz);
+    #elif defined(USE_CERT_BUFFERS_1024)
+        byte        key[sizeof_client_key_der_1024];
+        byte        cert[sizeof_client_cert_der_1024];
+        word32      keySz = (word32)sizeof(key);
+        word32      certSz = (word32)sizeof(cert);
+        XMEMSET(key, 0, keySz);
+        XMEMSET(cert, 0, certSz);
+        XMEMCPY(key, client_key_der_1024, keySz);
+        XMEMCPY(cert, client_cert_der_1024, certSz);
+    #else
+        unsigned char   cert[ONEK_BUF];
+        unsigned char   key[ONEK_BUF];
+        FILE*           fp;
+        int             certSz;
+        int             keySz;
+
+        fp = fopen("./certs/client_cert_der_1024", "rb");
+        AssertNotNull(fp);
+        certSz = fread(cert, 1, sizeof_client_cert_der_1024, fp);
+        fclose(fp);
+
+        fp = fopen("./certs/client_key_der_1024", "rb");
+        AssertNotNull(fp);
+        keySz = fread(key, 1, sizeof_client_key_der_1024, fp);
+        fclose(fp);
+    #endif
+
+    XMEMSET(output, 0, outputSz);
+    AssertIntEQ(wc_InitRng(&rng), 0);
+
+    AssertIntEQ(wc_PKCS7_InitWithCert(&pkcs7, cert, certSz), 0);
+
+    printf(testingFmt, "wc_PKCS7_VerifySignedData()");
+
+    pkcs7.content = data;
+    pkcs7.contentSz = (word32)sizeof(data);
+    pkcs7.privateKey = key;
+    pkcs7.privateKeySz = (word32)sizeof(key);
+    pkcs7.encryptOID = RSAk;
+    pkcs7.hashOID = SHAh;
+    pkcs7.rng = &rng;
+
+    AssertIntGT(wc_PKCS7_EncodeSignedData(&pkcs7, output, outputSz), 0);
+    wc_PKCS7_Free(&pkcs7);
+    AssertIntEQ(wc_PKCS7_InitWithCert(&pkcs7, NULL, 0), 0);
+    AssertIntEQ(wc_PKCS7_VerifySignedData(&pkcs7, output, outputSz), 0);
+
+    /* Test bad args. */
+    AssertIntEQ(wc_PKCS7_VerifySignedData(NULL, output, outputSz), BAD_FUNC_ARG);
+    AssertIntEQ(wc_PKCS7_VerifySignedData(&pkcs7, NULL, outputSz), BAD_FUNC_ARG);
+    AssertIntEQ(wc_PKCS7_VerifySignedData(&pkcs7, badOut,
+                                badOutSz), BAD_FUNC_ARG);
+
+    printf(resultFmt, passed);
+
+    wc_PKCS7_Free(&pkcs7);
+    wc_FreeRng(&rng);
+#endif
+} /* END test_wc_PKCS7_VerifySignedData() */
+
+
+/*
+ * Testing wc_PKCS7_EncodeEnvelopedData()
+ */
+static void test_wc_PKCS7_EncodeDecodeEnvelopedData (void)
+{
+#if defined(HAVE_PKCS7)
+    PKCS7       pkcs7;
+    word32      tempWrd32   = 0;
+    byte*       tmpBytePtr = NULL;
+    const char  input[] = "Test data to encode.";
+    int         i;
+    int         testSz = 0;
+    #if !defined(NO_RSA) && !defined(NO_AES) && (!defined(NO_SHA) |\
```

Comment:
There are several places that you are using the bitwise "|" operator between "if defined() | defined" statements.  We should be using the logical "||" operator instead.  Please adjust, thanks.